lib/database_cleaner/safeguard.rb

```diff
@@ -0,0 +1,72 @@
+module DatabaseCleaner
+  class Safeguard
+    class Error < Exception
+      class RemoteDatabaseUrl < Error
+        def initialize
+          super("ENV['DATABASE_URL'] is set to a remote URL. Please refer to https://github.com/DatabaseCleaner/database_cleaner#safeguards")
+        end
+      end
+
+      class ProductionEnv < Error
+        def initialize(env)
+          super("ENV['#{env}'] is set to production. Please refer to https://github.com/DatabaseCleaner/database_cleaner#safeguards")
+        end
+      end
+    end
+
+    class RemoteDatabaseUrl
+      LOCAL = %w(localhost 127.0.0.1)
```

Comment:
Maybe it's a good idea to also include `::1` for IPv6?

(also, all `127.0.0.0/8` addresses are local, although not sure if it's worth the effort adding support for that, as it's not often used)

Reply:
I'm sure PRs are welcome :)

Reply:
Definitely. However, I'd encourage you to file an issue first. In my opinion, it has to be a concrete issue first in order to get patched later.